articles/introduction-to-javascript-modules/index.md

```diff
@@ -0,0 +1,148 @@
+## Introduction to Javascript Modules
+
+As you write programs, once in a while, you'll find yourself in a position where you need to write several lines of code to achieve a specific task.  
+
+Instead of writing all these codes in a single file, you could as well separate concerns, by creating several files each with its own defined task.  
+
+With this approach, your code will be well organized and maintainable in the essence that in the future when you or even another developer wants to update the code, it becomes easier.
+
+
+### Getting Started With JavaScript Modules
+In Javascript, there is a functionality known as modules, this feature allows us to divide our Javascript codes into smaller portions.  
+This could enhance system security by enabling you to expose some parts of the code to be exported while keeping others.  
+
+A point to note, Javascript modules concept depends on the ```import``` and ```export``` statements.  
+
+### Export Module Feature
+Let's create a simple file and name it 
+```add.js```.
+```js
+// exporting a function
+export function addTwoNumbers(firstNumber,secondNumber) {
+    const sum=firstNumber+secondNumber;
+    return sum;
+}
+```
+In this file, we have a method(function) called ```addTwoNumbers```, which takes two parameters, ```firstNumber``` and ```lastNumber```.  
+But have you noticed the use of the keyword ```export``` coming before the function name?  
+
+What it does is very simple, it makes your code available for importation into another file. For example, the code in `add.js` is made available for use in another file.
+
+This means that the method defined here, ```addTwoNumbers()``` will be made available into the code where this file is imported.  
+
+The most common way of exporting items in Javascript has been the use of a single export statement at the end of the module file.  
+This is then followed by a list of items to be exported separated by commas in curly braces. For example:-  
+
+```code snippet for student.js```
+
+```js
+
+let studentName = 'John Doe';
+
+function studentDetails(name,
+                        admission_number,
+                        college,yearOfStudy
+)
+{
+    const myDetails="My name is "+ name
+     +", a student at " + college
+     +", admission number "+ admission_number
+     +". I'm in year"+ yearOfStudy;
+
+    return myDetails;
+}
+
+export { studentName,studentDetails };
+
+```
+In the above code, we have a method named ```studentDetails()```  which is being passed several parameters and, a variable named ```studentName```.  
```

Comment:
The comma goes before the word "and".

Mentioning the parameters of the function seems irrelevant.

```suggestion
In the above code, we have a method named ```studentDetails()```, and a variable named ```studentName```.